frontend/views/GameDetailedView.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { AppStore } from '../storage/common';
+import { Game } from '../storage/models/Game';
+import { GameSeries } from '../storage/models/GameSeries';
+import { Image } from '../storage/models/Image';
+import { Narrow } from '../components/styles/Narrow';
+import { PageHeader } from '../components/styles/PageHeader';
+import { PageTitle } from '../components/styles/PageTitle';
+import { PageSubtitle } from '../components/styles/PageSubtitle';
+import { PageSection } from '../components/styles/PageSection';
+import { SectionHeader } from '../components/styles/SectionHeader';
+import { GameBox } from '../components/GameBox';
+import { RouteComponentProps } from 'react-router-dom';
+
+interface MatchParams {
+	chosenGame: string;
+}
+interface GameDetailevViewProps {
+	games: Map<number, Game>;
+	gameSeries: Map<number, GameSeries>;
+	images: Map<number, Image>;
+	chosenGameId: number;
+}
+interface GameDetailedViewState {
+	chosenGame: Game;
+}
+
+export class GameDetailedView extends React.Component<GameDetailevViewProps, GameDetailedViewState> {
+	chosenSeries: GameSeries;
+	chosenImage: Image;
+	constructor(props: GameDetailevViewProps) {
+		super(props);
+		this.state = {
+			chosenGame:
```

Comment:
What I'd do instead is make it so that if there is no game matching for the ID the user is immediately rerouted to game-list page. This will save you from having to add the clunky fallback.

This might have more info: https://www.reddit.com/r/reactjs/comments/7laeel/redirect_before_componentdidmount/